src/django_components/util/template_tag.py

```diff
@@ -180,64 +69,54 @@ class TagParam:
     value: Any
 
 
-class TagParams(NamedTuple):
```

Comment:
`TagParams` can be removed as a class. Previously it was meant only to ensure that the `params`, and `tag_spec` were held in a joint object. But now that's done by `BaseNode`.

So, similarly to validation, I refactored `TagParams.resolve()` into a standalone function `resolve_params`